docs/src/main/asciidoc/config-reference.adoc

```diff
@@ -79,17 +79,33 @@ nor `\_` with `_`: `FOO_BAR_0_` or `FOO_BAR_0__BAZ`
 
 [CAUTION]
 ====
-In some situations, looking up the exact property name is impossible. For instance, when looking
-up a configuration that is part of a `Map`, and the property name contains a dynamic segment (the `Map` key). In this
-case, Quarkus relies upon each source’s list of property names. These must be converted back to their most likely
-dotted format for Environment Variables.
+In some situations, looking up the exact property name is impossible. This is the case for configuration names that
+contain user defined path segments. For instance:
```

Comment:
Not sure about the `For instance:` here as you start another sentence just below?